### ISSUE TITLE
Place labels in Uganda

### ISSUE THREAD

**Issue by wipfli** [protomaps/basemaps#194]
Place labels in Uganda at high zooms look all the same. It is hard to differentiate between important cities and tiny hamlets.

https://maps.protomaps.com/#map=11.26/-1.2703/29.9839&theme=light&renderer=maplibregl&tiles=https://build.protomaps.com/20231226.pmtiles

I think it is a problem in OSM because organic maps also looks terrible in Uganda.

**Comment by bdon** [comment]
Agreed... unlikely we have NE as a fallback for this level of detail. Can you investigate if any of those places have Wikidata tags, and if so is QRank a realistic option for ranking?

**Comment by wipfli** [comment]
![image](https://github.com/protomaps/basemaps/assets/53421382/3e0cee14-fce5-470f-849b-a96dc1428ba5)

Important cities seem to have wikidata in Uganda.

**Comment by wipfli** [comment]
Here is how a QRank based map looks like: https://wipfli.github.io/swiss-map/#map=11.26/-1.2703/29.9839

**Comment by nvkelso** [comment]
@wipfli want to submit a PR to add QRank to the populated places layer here in Protomaps? It looks like your example repo is a custom fork with a few other things going on...

I imagine a PR here would wire in qrank generally to the project and specifically in the places layer add another sort param (getSortKey function) in the existing label grid logic. So absent population information, we'd fallback to qrank before name length?

See also: 

- https://github.com/protomaps/basemaps/blob/main/tiles/src/main/java/com/protomaps/basemap/layers/Places.java#L34-L50

**Comment by wipfli** [comment]
Thanks for asking! The code in wipfli/swiss-map was actually written independently from scratch - and hence is a bit of a mess... I am happy to incorporate QRank for label hierarchy at some point in the future, but will first look into language customization.

**Comment by bdon** [comment]
We have QRank data being loaded but not used yet in our profile, so there's something to work off of.

**Comment by bdon** [comment]
@migurski this issue is also related to places density at low zooms
